### ISSUE TITLE
Postgres enums within schema

### ISSUE THREAD

**Issue by blankhart** [sfackler/rust-postgres#627]
The derived `FromSql` instance appears to fail when translating a Rust type to a Postgres type qualified by a schema name.  

For example, this works fine:

```rust
  #[derive(Clone, Copy, Debug, PartialEq, Eq, Deserialize, Serialize, FromSql, ToSql)]
  #[postgres(name = "service")]
  pub enum Service {
    Facebook,
    Google,
    Twitter,
  }
```

But this does not, the only difference being the "users." schema prefix:

```rust
  #[derive(Clone, Copy, Debug, PartialEq, Eq, Deserialize, Serialize, FromSql, ToSql)]
  #[postgres(name = "users.service")]
  pub enum Service {
    Facebook,
    Google,
    Twitter,
  }
```

The Postgres types in each case would correspond, i.e.:

```pgsql
CREATE TYPE [users.]service AS ENUM (
  'Facebook',
  'Google',
  'Twitter'
);
```

Perhaps though there is a way to use the qualified Postgres type that I've overlooked?

**Comment by sfackler** [comment]
The derive currently just ignores the schema of the type entirely I believe. Does the first code sample (with no schema qualification) not work with the type defined in a non-default schema?

**Comment by blankhart** [comment]
That works, yes, thanks!  The behavior is perhaps confusing if the schema is accepted in some cases but produces errors in others.  For example, `pg_mapper` accepts the schema.  If it's too much to have consistency across the related packages it may be helpful to add a note to the docs for `#[postgres]`.

**Comment by sfackler** [comment]
Yeah I think it is a bit weird that the logic currently ignores the schema entirely. I think it'd definitely make sense to handle the case where the type name is schema-qualified and add a check for the schema name at that point. Would you be interested in making a PR?

**Comment by blankhart** [comment]
Yes but I am new to Rust and databases and quite time constrained to if this is waiting on me it may be a while.

**Comment by sfackler** [comment]
No worries!

**Comment by mfwre** [comment]
Was this worked on?
